tests/test_stack.py

```diff
@@ -849,6 +850,16 @@ def test_get_role_arn_with_role(self):
         self.stack.config["role_arn"] = sentinel.role_arn
         assert self.stack._get_role_arn() == {"RoleARN": sentinel.role_arn}
 
+    def test_protect_execution_without_protection(self):
+        self.stack._config = {"protect": False}
+        # Function should do nothing if protect == False
+        self.stack._protect_execution()
+
+    def test_protect_execution_with_protection(self):
+        self.stack._config = {"protect": True}
+        with pytest.raises(ProtectedStackError):
+            self.stack._protect_execution()
+
```

Comment:
Can we add a test where "protect" isn't in the config, as thats typical for most stacks.